SDK.md

```diff
@@ -0,0 +1,281 @@
+ - `body` is a stream, since CloudEvent spec does not define exact type of an 
+    event data, SDK should not be responsible for data coercing
+ - `data_unmarshaller` is a function that performs data unmarshaller, 
+    logic of this method may vary depending on the type of an event format
+ - the return statement is a CloudEvent of the particular spec
+
+#### HTTP API marshaller
+
+An HTTP marshaller is a method that should be capable to convert a CloudEvent 
+into a combination of an HTTP request headers and a body.
+Here's the signature of a marshaller:
+```
+    ToRequest(
+        event CloudEvent,
+        converter_type FormatRef,
+        data_marshaller function(data serializable) -> object
+    ) -> map[string]string, Stream :
+```
+
+In this signature:
+
+ - `event` is a CloudEvent of the particular spec
+ - `converter_type` represents a type of an HTTP binding format 
+    ([binary](http-transport-binding.md#31-binary-content-mode) or 
+     [structured](http-transport-binding.md#32-structured-content-mode))
+ - `data_marshaller` is a function that serialized an event data
+ - the return statement is a set of an HTTP headers and request body
+
+The reason for such signature is that in most of programming languages 
+there are a lot if different HTTP frameworks and route handling signature may vary, 
+but HTTP headers and a request body is common 
+(or may be easily converted to the appropriate type).
+
+
+#### HTTP Converters API
+
+Each transport binding unmarshaller/marshaller is a set of converters.
+In terms of an SDK the converter is a set API methods that are actually converting 
+a CloudEvent of the particular spec into a transport binding-ready data.
+
+Converter API consists of at least two methods:
+```
+    Read(...)
+    Write(...)
+```
+
+Signature of these methods may vary depending on the the type of a transport binding.
+
+
+HTTP Converters API is the example of Converters API implementation 
+for the [HTTP transport binding](http-transport-binding.md).
+Taking into account that the CloudEvent spec defines 2 (structured and binary) formats,
+an SDK suppose to implement two converters (one per each format):
+
+ - `BinaryHTTPCloudEventConverter`
+ - `StructuredHTTPCloudEventConverter`
+
+HTTP Converters suppose to comply the following signature:
+```
+    Read(
+        event CloudEvent,
+        headers headers HTTP-Headers,
+        body Stream,
+        data_unmarshaller function(data serializable) -> object
+    ) -> CloudEvent
+    
+    Write(
+        event CloudEvent,
+        data_marshaller function(data serializable) -> object
+    ) -> HTTP-Headers, Stream
+    
+```
+
+As you may see, Converter signature follows the signature of marshaller/unmarshaller.
+It means that an HTTP marshaller/unmarshaller is a set of binary and structured converters.
+
+In `Read`:
+
+ - `event` - a placeholder (empty event object), see [event constructor](#Event object constructor API).
```

Comment:
I think `#Event object constructor API` needs to be `#event-object-constructor-api`

Reply:
fixed, thanks